### PR TITLE
ui5 tooling v3

### PR DESCRIPTION
**Pull request by mauriciolauffer**
ui5 tooling v3 + better `eslint` rules.

I was going to remove almost everything from `eslint`, but I thought it'd be simpler to just make minor changes.

This'd be the proposed rules:

```yaml
root: true
extends:
  - eslint:recommended
env:
  browser: true
  es6: true
  jquery: true
  qunit: true
globals:
  sap: true
  sinon: true
rules:
  no-var: error
```

### CODE REVIEW

File: webapp/manifest.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-	"_version": "1.12.0",
+	"_version": "1.56.0",
```
Comment (d3xter666) [inline webapp/manifest.json:2]: Are we looking for the latest schema version here?
Then it should be `1.57.0`. Please, check the spec here: https://raw.githubusercontent.com/SAP/ui5-manifest/master/schema.json

**Review by d3xter666 — COMMENTED** [review]
Overall, I'd assume that those replacements of `var` should cover the scope of the whole application and also most if not all of them should be replaced by `const` instead of `let`.

For the `eslint` rules, I'd request collesgues to give their feedback

**Comment by flovogt** [comment]
@mauriciolauffer Thanks a lot for this PR. We're planning to transform this sample app to more modern ECMAScript syntax and will consider this PR also when doing the transformation.

File: webapp/controller/App.controller.js [hunk 2]
```diff
@@ -39,12 +39,12 @@ sap.ui.define([
 		 * Removes all completed items from the todo list.
 		 */
 		clearCompleted: function() {
-			var oModel = this.getView().getModel();
-			var aTodos = oModel.getProperty("/todos").map(function (oTodo) { return Object.assign({}, oTodo); });
+			let oModel = this.getView().getModel();
```
[the remark this answers is not in the document]
Comment (mauriciolauffer) [reply webapp/controller/App.controller.js:42]: I've just applied `eslint --fix`, I haven't manually changed the declarations...

**Pull request closed without merging**